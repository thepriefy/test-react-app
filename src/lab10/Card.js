import React from 'react'

const Card = (props) => {
    return (
        <div>
            <h3 className="card_header">{props.header_text}</h3>
            <p className="card_content">{props.content_text}</p>
            <small className="card_footer">{props.footer_text}</small>
        </div>
    )
}

export default Card