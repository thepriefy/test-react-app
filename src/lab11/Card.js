import React from 'react'

const Card = (props) => {
    return (
        <div>
            <h3 className="card_header">{props.header_text}</h3>
            <p className="card_content">{props.content_text}</p>
            <button className="card_btn" onClick={props.onClick}>click!</button>
        </div>
    )
}

export default Card