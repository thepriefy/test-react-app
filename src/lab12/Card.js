import  React from 'react'

class Card extends React.Component {
    state = {
        title: 'This is title.',
        item_list: []
    }

    render () {
        return (
            <div>
                <h1>{this.state.title}</h1>
                <h3 className="card_header">{this.props.header_text}</h3>
                <p className="card_content">{this.props.content_text}</p>
                <button className="card_btn" onClick={this.props.onClick}>click!</button>
            </div>
        )
    }
}

export default Card