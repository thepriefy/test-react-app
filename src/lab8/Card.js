import React from 'react'

const Card = () => {
    return (
        <div>
            <h3 className="header">Card Header</h3>
            <p className="content">Content</p>
            <button className="action_btn">action!</button>
        </div>
    )
}

export default Card