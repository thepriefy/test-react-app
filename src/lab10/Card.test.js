import React from 'react'
import sinon from 'sinon'
import {shallow} from 'enzyme'
import Card from './Card'

describe('Test Card component with props', () => {
    it('Match it snapshot', () => {
        const props = {
            header_text: 'This is header',
            content_text: 'This is content',
            footer_text: 'This is footer'
        }

        const wrapper = shallow(<Card {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })
})