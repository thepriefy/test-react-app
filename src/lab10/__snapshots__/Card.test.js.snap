// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component with props Match it snapshot 1`] = `
<div>
  <h3
    className="card_header"
  >
    This is header
  </h3>
  <p
    className="card_content"
  >
    This is content
  </p>
  <small
    className="card_footer"
  >
    This is footer
  </small>
</div>
`;
